src/core/network/createNetwork.js

```diff
@@ -12,61 +12,91 @@ governing permissions and limitations under the License.
+     * @property {Object} payload Payload object that will be sent.
      */
     /**
-     * Create a new request.  Once "send" on the returned object is called, the lifecycle
-     *  method "onBeforeSend" will be triggered with { payload, responsePromise } as
-     *  the parameter.  When the response is returned it will call the lifecycle method "onResponse"
-     *  with the returned response object.
+     * Create a new request.  Once "send" on the returned object is called, the
+     * lifecycle method "onBeforeSend" will be triggered with
+     * { payload, responsePromise, isBeacon } as the parameter.  When the
+     * response is returned it will call the lifecycle method "onResponse"
+     * with the returned response object.
+     *
      * @param {boolean} [expectsResponse=true] The endpoint and request mechanism
      * will be determined by whether a response is expected.
      * @returns {Request}
      */
     newRequest(expectsResponse = true) {
       const payload = createPayload();
-      const action = expectsResponse ? "interact" : "collect";
-      const url = `https://${edgeDomain}/${action}?propertyID=${propertyID}`;
-      const deferred = defer();
-      const responsePromise = deferred.promise
-        .then(() =>
-          lifecycle.onBeforeSend({
-            payload,
-            responsePromise,
-            isBeacon: expectsResponse
+
+      const send = () => {
+        const responsePromise = Promise.resolve()
+          .then(() =>
+            lifecycle.onBeforeSend({
+              payload,
+              responsePromise,
+              isBeacon: expectsResponse
+            })
+          )
+          .then(() => {
+            const action = expectsResponse ? "interact" : "collect";
+            const url = `https://${edgeDomain}/${action}?propertyID=${propertyID}`;
+            const responseHandlingMessage = expectsResponse
+              ? ""
+              : " (response will be ignored)";
+            logger.log(
+              `Sending network request${responseHandlingMessage}:`,
+              payload.toJSON()
+            );
+
+            return networkStrategy(
+              url,
+              JSON.stringify(payload),
+              expectsResponse
+            );
           })
-        )
-        .then(() => {
-          const networkResponse = networkStrategy(
-            url,
-            JSON.stringify(payload),
-            expectsResponse
-          );
+          .then(responseBody => {
+            let handleResponsePromise;
+
+            if (expectsResponse) {
+              handleResponsePromise = handleResponse(responseBody);
+            }
 
-          return networkResponse;
-        });
+            return handleResponsePromise;
+          });
 
-      if (expectsResponse) {
-        responsePromise.then(handleResponse);
-      }
+        return responsePromise;
+      };
 
-      return { payload, responsePromise, send: deferred.resolve };
```

Comment:
We no longer return a `responsePromise` when `newRequest` is called. It made more sense to me to return the promise from `send` when it is called.